lib/set.js

```diff
@@ -6,8 +6,36 @@
  * @param {*} value
  *
  */
-function set (obj_, path, value) {
-  throw "Still a skeleton";
+
+function set (obj, path, value) {
+  if (obj == null) {
+    return obj;
+  }
+
+  if (typeof path === "string") {
+    path = path.split(".");
+  }
+
+  if (!Array.isArray(path)) {
+    throw "Path should be a string or array";
+  }
+
+  let node = obj;
+  const bound = path.length - 1;
+
+  let i, key;
+  for (i = 0; i < bound; i += 1) {
+    key = path[i];
+    if (node[key] === undefined) {
+      node[key] = {};
+    }
+    node = node[key];
+  }
+
+  key = path[i];
+  node[key] = value;
+
+  return obj;
```

Comment:
Could you maybe explain how this is working? I'm a little confused that since you are returning the original `obj`parameter without mutating it. thanks!

Reply:
Sure! Mutation is actually being applied to the obj parameter.

See L23 where obj is assigned to the `node` variable.

We then walk the path updating the node variable to be the current point in the path. If the current node doesn't have that point in the path, we insert it.

Finally, for the final point in the path, we assign the value.

Reply:
Ah! Very cool. Thanks @SamHowie !! Gonna go ahead and merge this in. Thanks!